dotnet/Services/CybersourcePaymentService.cs

```diff
@@ -565,15 +566,29 @@ public async Task<Payments> BuildPayment(CreatePaymentRequest createPaymentReque
                         foreach (PriceTag priceTag in vtexOrderItem.PriceTags)
                         {
                             string name = priceTag.Name.ToLower();
-                            if ((name.Contains("tax@") || name.Contains("taxhub@")) && !name.Contains("shipping"))
+                            if (name.Contains("tax@") || name.Contains("taxhub@"))
                             {
-                                if (priceTag.IsPercentual ?? false)
+                                if (name.Contains("shipping"))
                                 {
-                                    itemTax += (long)Math.Round(vtexOrderItem.SellingPrice * priceTag.RawValue, MidpointRounding.AwayFromZero);
+                                    if (priceTag.IsPercentual ?? false)
+                                    {
+                                        shippingTax += (long)Math.Round(vtexOrderItem.SellingPrice * priceTag.RawValue, MidpointRounding.AwayFromZero);
```

Comment:
`vtexOrderItem.SellingPrice` is incorrect ... the tax rate needs to be applied to the shipping cost, not the item cost. Maybe you can just store the shipping tax rate here, and then use it to calculate the tax when you build the shipping tax line item?